From: Anibal Monsalve Salazar <anibal@debian.org>
Date: Wed, 20 Nov 2019 13:08:40 -0500
Subject: no-cpio.info-2

---
 doc/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/doc/Makefile.am b/doc/Makefile.am
index 98d26da..b685d60 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -17,7 +17,6 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
 # 02110-1301 USA.
 
-info_TEXINFOS = cpio.texi
 dist_man_MANS = cpio.1 $(MT_1) $(RMT_8)
 
 if CPIO_MT_COND
