Description: fix date-sensitive test
Author: Marc Deslauriers <marc.deslauriers@canonical.com>

Index: cpio-2.11+dfsg/tests/symlink-bad-length.at
===================================================================
--- cpio-2.11+dfsg.orig/tests/symlink-bad-length.at	2016-02-18 10:36:51.066541370 -0500
+++ cpio-2.11+dfsg/tests/symlink-bad-length.at	2016-02-18 10:36:51.066541370 -0500
@@ -57,7 +57,7 @@
 echo >&2 STDERR
 ],
 [0],
-[-rw-rw-r--   1 10029    10031          13 Nov 25 11:52 FILE
+[-rw-rw-r--   1 10029    10031          13 Nov 25  2014 FILE
 ],[STDERR
 ])
 
Index: cpio-2.11+dfsg/tests/testsuite
===================================================================
--- cpio-2.11+dfsg.orig/tests/testsuite	2016-02-18 10:36:51.030541004 -0500
+++ cpio-2.11+dfsg/tests/testsuite	2016-02-18 10:37:18.338816966 -0500
@@ -2172,7 +2172,7 @@
 echo >>"$at_stderr"; $as_echo "STDERR
 " | \
   $at_diff - "$at_stderr" || at_failed=:
-echo >>"$at_stdout"; $as_echo "-rw-rw-r--   1 10029    10031          13 Nov 25 11:52 FILE
+echo >>"$at_stdout"; $as_echo "-rw-rw-r--   1 10029    10031          13 Nov 25  2014 FILE
 " | \
   $at_diff - "$at_stdout" || at_failed=:
 at_fn_check_status 0 $at_status "$at_srcdir/symlink-bad-length.at:46"
