From: Anibal Monsalve Salazar <anibal@debian.org>
Date: Wed, 20 Nov 2019 13:08:40 -0500
Subject: Return MT_EXIT_FAILURE instead of MT_EXIT_INVOP for fatal exits from
 mt.

---
 src/Makefile.am | 2 ++
 src/mt.c        | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/src/Makefile.am b/src/Makefile.am
index ab3fc1a..6430587 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -42,6 +42,8 @@ cpio_SOURCES = \
  makepath.c\
  userspec.c
 
+mt_SOURCES = mt.c
+
 noinst_HEADERS =\
  cpio.h\
  cpiohdr.h\
diff --git a/src/mt.c b/src/mt.c
index 7022738..1cc23b1 100644
--- a/src/mt.c
+++ b/src/mt.c
@@ -55,6 +55,8 @@
 
    David MacKenzie <djm@gnu.ai.mit.edu> */
 
+#include "configmake.h"
+
 #include <system.h>
 
 #include <stdio.h>
