From: Anibal Monsalve Salazar <anibal@debian.org>
Date: Wed, 20 Nov 2019 13:08:40 -0500
Subject: Return MT_EXIT_FAILURE instead of MT_EXIT_INVOP for fatal exits from
 mt.

---
 src/Makefile.am | 2 ++
 src/mt.c        | 2 ++
 2 files changed, 4 insertions(+)

Index: cpio-2.14+dfsg/src/Makefile.am
===================================================================
--- cpio-2.14+dfsg.orig/src/Makefile.am
+++ cpio-2.14+dfsg/src/Makefile.am
@@ -41,6 +41,8 @@ cpio_SOURCES = \
  makepath.c\
  userspec.c
 
+mt_SOURCES = mt.c
+
 noinst_HEADERS =\
  cpio.h\
  cpiohdr.h\
Index: cpio-2.14+dfsg/src/mt.c
===================================================================
--- cpio-2.14+dfsg.orig/src/mt.c
+++ cpio-2.14+dfsg/src/mt.c
@@ -54,6 +54,8 @@
 
    David MacKenzie <djm@gnu.ai.mit.edu> */
 
+#include "configmake.h"
+
 #include <system.h>
 
 #include <stdio.h>
